Mathlib/LinearAlgebra/Charpoly/ToMatrix.lean

```diff
@@ -22,9 +22,12 @@ of `f` in any basis.
 
 universe u v w
 
-variable {R : Type u} {M : Type v} [CommRing R] [Nontrivial R]
-
-variable [AddCommGroup M] [Module R M] [Module.Free R M] [Module.Finite R M] (f : M →ₗ[R] M)
+variable {R : Type u} {M : Type v} {M₁ M₂ : Type*} [CommRing R] [Nontrivial R]
+variable [AddCommGroup M] [AddCommGroup M₁] [AddCommGroup M₂]
+variable [Module R M] [Module R M₁] [Module R M₂]
+variable [Module.Free R M] [Module.Finite R M₁] [Module.Finite R M₂]
+variable [Module.Finite R M] [Module.Free R M₁] [Module.Free R M₂]
+variable (f : M →ₗ[R] M) (f₁ : M₁ →ₗ[R] M₁) (f₂ : M₂ →ₗ[R] M₂)
```

Comment:
This is a lot of variables to have for just one declaration. Maybe just add them to that declaration instead?

Reply:
Ok, done. I did leave the types `M_1` and `M_2` in place, so that they appear at the beginning of the list of parameters, instead of after the `[CommRing R] [Nontrivial R]`, which I prefer.